ddapm_test_agent/agent.py

```diff
@@ -329,23 +331,42 @@ async def _handle_traces(self, request: Request, version: Literal["v0.4", "v0.5"
         await self._store_request(request)
         token = request["session_token"]
         checks: Checks = request.app["checks"]
+        headers = CIMultiDict(request.headers)
 
-        await checks.check("trace_stall", headers=request.headers, request=request)
+        log.debug(f"New request: {request} with headers: {headers}")
+        log.debug(f"Request Data: {self._request_data(request)!r}")
+
+        await checks.check("trace_stall", headers=headers, request=request)
+
+        proxy_to_agent = True
+        if "Do-Not-Proxy-To-Agent" in headers:
```

Comment:
```suggestion
        if "Datadog-Agent-Proxy-Enabled" in headers:
```

Can we document this here https://github.com/DataDog/dd-apm-test-agent#proxy

also add a release note

ideally this is its own change